app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -732,13 +761,27 @@ private void handleInstallButtonClick() {
         });
   }
 
-  private void showRecommendsDialog(boolean isLoggedIn, String packageName) {
+  private void handleDialogImpressions() {
+    view.getLifecycle()
+        .filter(lifecycleEvent -> lifecycleEvent.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> dialogImpression)
+        .filter(dialogImpression -> dialogImpression)
+        .observeOn(Schedulers.io())
+        .flatMap(
+            __ -> appViewManager.recordABTestImpression(ABTestManager.ExperimentType.SHARE_DIALOG))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void showRecommendsDialog(boolean isLoggedIn, String packageName, Experiment experiment) {
     if (isLoggedIn && appViewManager.shouldShowRecommendsPreviewDialog()) {
-      view.showRecommendsDialog();
+      view.showRecommendsDialog(experiment);
       appViewAnalytics.sendLoggedInRecommendAppDialogShowEvent(packageName);
+      dialogImpression.onNext(true);
```

Comment:
We shouldn't use the publish subject here to "communicate" the impression. I think we can turn this method an observable and make the impression request in the chain that calls this.

Reply:
Not really fixed but was discussed